### PR TITLE
feat(invoices): return the paid to date and balance in the invoices endpoint

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Returns the PaidToDate and Balance properties in the invoices endpoint:

image

**Changes:**

- Added the PaidToDate and Balance properties in the InvoiceListItem object and return these in the invoices endpoint.

- Added an unit test to validate the balance is calculated correctly  .

**Task:** [DAT-41](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-41)

### CODE REVIEW

File: Billing.API/Models/InvoiceListItem.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ public class InvoiceListItem
         public DateTimeOffset Date { get; }
         public string Currency { get; }
         public double Amount { get; }
+        public double PaidToDate { get; }
+        public double Balance { get { return Amount - PaidToDate; } }
```
Comment (nreal) [inline Billing.API/Models/InvoiceListItem.cs:18]: Is possible that the Balance can be negative?

Comment (jhoffman-ms) [reply Billing.API/Models/InvoiceListItem.cs:18]: Should not. Why?

Comment (nreal) [reply Billing.API/Models/InvoiceListItem.cs:18]: Because PaidTodate can be major that amount but is a really rare case

Comment (jhoffman-ms) [reply Billing.API/Models/InvoiceListItem.cs:18]: Exactly

**Comment by fgchaio** [comment]
👍

File: Billing.API/Models/InvoiceListItem.cs (hunk 1 above)
Comment (swidzinskiMS) [reply Billing.API/Models/InvoiceListItem.cs:18]: We do emit credit notes, maybe that should be shown as a negative balance... not sure but you could ask Noe

Comment (msimionati) [reply Billing.API/Models/InvoiceListItem.cs:18]: I have some doubts regarding to the meaning of the Balance and the PaidToDate attributes:
do these values ​​represent the invoice item level? 
Can an item be partially paid?

Comment (jhoffman-ms) [reply Billing.API/Models/InvoiceListItem.cs:18]: On monday we can to discuss it in our call.

Comment (jhoffman-ms) [reply Billing.API/Models/InvoiceListItem.cs:18]: Noe's comment: "Se puede dar cuando haya Notas de crédito o cuando el cliente pague por demás. Pasa...se espera que salga en negativo.". It behavior works fine in the current implementation:

image

Comment (swidzinskiMS) [reply Billing.API/Models/InvoiceListItem.cs:18]: Great

**Comment by jhoffman-ms** [comment]
Do you have another feedback or suggestion or is ready to merge?

**Comment by nreal** [comment]
👍

**Review by bamaro-dev — COMMENTED** [review]
✔️

**Pull request merged**